download/download.go

```diff
@@ -183,17 +193,17 @@ func (d *Downloader) loop(ctx context.Context) {
 
 func (d *Downloader) oneShot(ctx context.Context) (bool, error) {
 	m := metrics.New()
-	b, etag, longPoll, err := d.download(ctx, m)
+	b, etag, longPoll, r, err := d.download(ctx, m)
```

Comment:
💭 Time for a struct, maybe?

Reply:
I was trying to push the limits here 😉 . Updated with a struct.